test/unit/relayServerUtils.test.ts

```diff
@@ -161,46 +168,102 @@ describe('relayServerUtils tests', function () {
 
     describe('When is not sponsored', function () {
       describe('Transfer fee scenarios', function () {
-        it('Should charge transferFee when a transfer is being relayed', async function () {
-          const request = createRequest({
-            data: dataWhenTransfer,
+        describe('Token', function () {
+          it('Should charge transferFee when a transfer is being relayed', async function () {
+            const request = createRequest({
+              data: dataWhenTransfer,
+              tokenContract: fakeToAddress,
+            });
+
+            const config = {
+              disableSponsoredTx: true,
+              sponsoredDestinations: [],
+              gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+              transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
+            } as unknown as AppConfig;
+
+            const fee = await relayServerUtils.calculateFee(
+              request,
+              BigNumber.from(FAKE_ESTIMATION_BEFORE_FEES),
+              config
+            );
+
+            expect(fee.toString()).to.eq(
+              expectedFeeFromTransferInTokenGas.toString()
+            );
           });
 
-          const config = {
-            disableSponsoredTx: true,
-            sponsoredDestinations: [],
-            gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
-            transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
-          } as unknown as AppConfig;
-
-          const fee = await relayServerUtils.calculateFee(
-            request,
-            BigNumber.from(FAKE_ESTIMATION_BEFORE_FEES),
-            config
-          );
-
-          expect(fee.toString()).to.eq(expectedFeeFromTransferInGas.toString());
+          it('Should charge transferFee when a transferFrom is being relayed', async function () {
+            const request = createRequest({
+              data: dataWhenTransferFrom,
+              tokenContract: fakeToAddress,
+            });
+
+            const config = {
+              disableSponsoredTx: true,
+              sponsoredDestinations: [],
+              gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+              transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
+            } as unknown as AppConfig;
+
+            const fee = await relayServerUtils.calculateFee(
+              request,
+              BigNumber.from(FAKE_ESTIMATION_BEFORE_FEES),
+              config
+            );
+
+            expect(fee.toString()).to.eq(
+              expectedFeeFromTransferInTokenGas.toString()
+            );
+          });
         });
 
-        it('Should charge transferFee when a transferFrom is being relayed', async function () {
-          const request = createRequest({
-            data: dataWhenTransferFrom,
+        describe('Native token', function () {
```

Comment:
```suggestion
        describe('Using native token', function () {
```
or 
```suggestion
        describe('Using RBTC', function () {
```

Reply:
I have some doubts here, should I standardize the name across the projects since I use native everywhere, should I use RBTC?

Reply:
updated